payments-core/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -236,7 +244,7 @@ internal class PaymentSheetViewModel internal constructor(
         val paymentSelection = selection.value
 
         if (paymentSelection is PaymentSelection.GooglePay) {
-            paymentIntent.value?.let { paymentIntent ->
+            (stripeIntent.value as? PaymentIntent)?.let { paymentIntent ->
```

Comment:
same here, since we're now expanding it to two intents, i think we should log the error when a `SetupIntent` incorrectly reaches here

Reply:
Google Pay is currently supported only for PaymentIntents, and that's enforced on `PaymentSheetContract.Args` where we set `isGooglePayEnabled` to false if it's not a PaymentIntent. So it'll always be a PaymentIntent here, we just need to cast it to pass as a parameter to `StripeGooglePayContract.Args`. Adding an error log anyways.